lib/spack/spack/test/views.py

```diff
@@ -23,3 +24,55 @@ def test_remove_extensions_ordered(install_mockery, mock_fetch, tmpdir):
 
     e1 = e2["extension1"]
     view.remove_specs(e1, e2)
+
+
+@pytest.mark.regression("32456")
+def test_view_with_spec_not_contributing_files(install_mockery, mock_fetch, tmpdir):
+    # This test is a bit contrived, but it checks whether
+    # pkg.add_files_to_view() receives the correct source paths
+    # when another package does not have any files to
+    # contribute to a view. That can happen either because
+    # there simply are no files in the prefix, or because all
+    # its files conflict with another spec -- here we test the
+    # former).
+
+    view_dir = str(tmpdir)
+    layout = DirectoryLayout(view_dir)
+    view = SimpleFilesystemView(view_dir, layout)
+
+    a = Spec("libelf").concretized()
+    b = Spec("trivial-install-test-package").concretized()
+    a.package.do_install()
+    b.package.do_install()
```

Comment:
I think this test could be simplified:

* Don't actually install the packages: mark them as concrete/installed, then assign the prefix to tmpdir (that way you don't need the below logic to clear out whatever was installed before)

Reply:
done